app/models/application_record.rb

```diff
@@ -1,3 +1,26 @@
 class ApplicationRecord < ActiveRecord::Base
   self.abstract_class = true
+
+  def self.get_by_key(key, params)
```

Comment:
I haven't seen class methods in the application controller before. This is really convenient for our purposes, and I'm happy with them here. If we have time, we should research if they can go (or belong) somewhere else.